# This is already fixed upstream. Remove this patch in the next integration.
diff --git a/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DecomposeUnsupportedConversions.cpp b/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DecomposeUnsupportedConversions.cpp
deleted file mode 100644
--- a/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DecomposeUnsupportedConversions.cpp
+++ /dev/null
@@ -1,100 +0,0 @@
-#include "TritonNVIDIAGPUToLLVM/Passes.h"
-#include "mlir/Pass/Pass.h"
-#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
-#include "triton/Analysis/Utility.h"
-#include "triton/Conversion/TritonGPUToLLVM/Patterns.h"
-#include "triton/Dialect/Triton/IR/Dialect.h"
-#include "triton/Dialect/TritonGPU/IR/Dialect.h"
-
-using namespace mlir;
-
-namespace mlir {
-namespace triton {
-#define GEN_PASS_DEF_DECOMPOSEUNSUPPORTEDNVIDIACONVERSIONS
-#include "TritonNVIDIAGPUToLLVM/Passes.h.inc"
-} // namespace triton
-} // namespace mlir
-
-namespace {
-
-using namespace mlir;
-using namespace triton;
-using namespace triton::gpu;
-
-// Loading from Hopper shared memory layout to dot operand is not supported. We
-// need to break it down and use a different shared layout. This would mostly
-// happen when TMAs are used with MMAV2 and will cause poor performance.
-class DecomposeLocalLoadToDotOperand
-    : public OpRewritePattern<triton::gpu::LocalLoadOp> {
-public:
-  using OpRewritePattern::OpRewritePattern;
-
-  LogicalResult matchAndRewrite(triton::gpu::LocalLoadOp op,
-                                PatternRewriter &rewriter) const override {
-
-    auto dstDotOp = dyn_cast<triton::gpu::DotOperandEncodingAttr>(
-        op.getType().getEncoding());
-    MemDescType srcType = op.getSrc().getType();
-    auto sharedEncoding = dyn_cast<SharedEncodingAttr>(srcType.getEncoding());
-    if (!dstDotOp || !sharedEncoding || !sharedEncoding.getHasLeadingOffset())
-      return failure();
-    RankedTensorType type = op.getType();
-    auto parentEnc = dstDotOp.getParent();
-    int numWarps = triton::gpu::getNumWarpsPerCTA(parentEnc);
-    int threadsPerWarp = triton::gpu::TritonGPUDialect::getThreadsPerWarp(
-        op->getParentOfType<ModuleOp>());
-    int numCTAs = triton::gpu::getNumCTAs(parentEnc);
-    auto blockEncoding = getDefaultBlockedEncoding(
-        op.getContext(), type.getShape(), numWarps, threadsPerWarp, numCTAs);
-    auto tmpType = RankedTensorType::get(type.getShape(), type.getElementType(),
-                                         blockEncoding);
-    Value load =
-        rewriter.create<LocalLoadOp>(op.getLoc(), tmpType, op.getSrc());
-    auto newSharedDescTy = MemDescType::get(
-        type.getShape(), type.getElementType(),
-        triton::gpu::SharedEncodingAttr::get(
-            op.getContext(), dstDotOp, type.getShape(),
-            triton::gpu::getOrder(parentEnc),
-            triton::gpu::getCTALayout(parentEnc), type.getElementType()),
-        srcType.getMemorySpace());
-    auto tmp = rewriter.create<triton::gpu::LocalAllocOp>(
-        op.getLoc(), newSharedDescTy, load);
-    auto newConvert =
-        rewriter.create<triton::gpu::LocalLoadOp>(op.getLoc(), type, tmp);
-    rewriter.replaceOp(op, newConvert);
-    return success();
-  }
-};
-
-struct DecomposeUnsupportedConversions
-    : public mlir::triton::impl::DecomposeUnsupportedNVIDIAConversionsBase<
-          DecomposeUnsupportedConversions> {
-  void runOnOperation() override {
-    // FIXME [Dot LL]
-    // Remove the decomposeTensorCoreToDotLayoutConversion class entirely after
-    // we have enabled the new layout conversion for all the cases.
-    auto nvidiaShortCutFn = [&](RankedTensorType srcTy,
-                                RankedTensorType dstTy) { return true; };
-    ModuleOp mod = getOperation();
-    triton::gpu::decomposeSplatOpToSharedLayoutConversion(mod);
-    triton::gpu::decomposeTensorCoreToDotLayoutConversion(mod,
-                                                          nvidiaShortCutFn);
-    triton::gpu::decomposeBlockedToDotLayoutConversion(mod);
-
-    mlir::RewritePatternSet patterns(&getContext());
-    patterns.add<DecomposeLocalLoadToDotOperand>(&getContext());
-    if (mlir::applyPatternsGreedily(mod, std::move(patterns)).failed()) {
-      signalPassFailure();
-    }
-  }
-};
-} // namespace
-
-namespace mlir::triton::NVIDIA {
-
-std::unique_ptr<OperationPass<ModuleOp>>
-createDecomposeUnsupportedConversionsPass() {
-  return std::make_unique<DecomposeUnsupportedConversions>();
-}
-
-} // namespace mlir::triton::NVIDIA
diff --git a/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv5.cpp b/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv5.cpp
--- a/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv5.cpp
+++ b/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv5.cpp
@@ -3,6 +3,7 @@
 #include "PatternTritonGPUOpToLLVM.h"
 #include "Utility.h"
 #include "mlir/Support/LLVM.h"
+#include "third_party/triton/include/triton/Conversion/TritonGPUToLLVM/Utility.h"
 #include "triton/Conversion/TritonGPUToLLVM/PatternTritonGPUOpToLLVM.h"
 
 using namespace mlir;
@@ -60,7 +61,8 @@ enum class mxfpKind { mxf8f6f4 = 0, mxf4
 inline mxfpKind getMXFPKind(ScaleDotElemType typeA, ScaleDotElemType typeB,
                             Type scaleAType, Type scaleBType) {
   if (typeA == ScaleDotElemType::E2M1 && typeB == ScaleDotElemType::E2M1) {
-    if (scaleAType.isFloat8E4M3FN() && scaleBType.isFloat8E4M3FN()) {
+    if (llvm::isa<mlir::Float8E4M3FNType>(scaleAType) &&
+        llvm::isa<mlir::Float8E4M3FNType>(scaleBType)) {
       return mxfpKind::mxf4nvf4;
     }
     return mxfpKind::mxf4;
@@ -100,10 +102,11 @@ static Value createInstDescriptor(Conver
       return 1;
     if (type.isF32())
       return 2;
-    if (type.isFloat8E4M3FN())
+    if (llvm::isa<mlir::Float8E4M3FNType>(type))
       return 0;
-    if (type.isFloat8E5M2())
+    if (llvm::isa<mlir::Float8E5M2Type>(type))
       return 1;
+
     llvm_unreachable("Unsupported type.");
   };
   static_assert(sizeof(TCGen5InstructionDescriptor) == 4,
@@ -224,7 +227,8 @@ static void createGen5MMA(ConversionPatt
     opcode += "f16";
   else if (srcElementTy.isF32())
     opcode += "tf32";
-  else if (srcElementTy.isFloat8E4M3FN() || srcElementTy.isFloat8E5M2())
+  else if (llvm::isa<mlir::Float8E4M3FNType>(srcElementTy) ||
+           llvm::isa<mlir::Float8E5M2Type>(srcElementTy))
     opcode += "f8f6f4";
   else
     assert(0 && "Unsupported type.");
